R/ResultFlagsDependent.R

```diff
@@ -548,6 +548,126 @@ TADA_FindQCActivities <- function(.data, clean = FALSE, flaggedonly = FALSE) {
 }
 
 
+#' Pair Replicates with Original Samples
+#' 
+#' This function looks for replicate samples and pairs them to their original or
+#' parent sample for further analysis. Replicate samples without an associated original
+#' sample are flagged as orphan samples.
+#' 
+#' @param .data TADA dataframe which must include the columns 'OrganizationIdentifier', 'ActivityTypeCode',
+#' 'ActivityStartDate', 'ActivityStartTime.Time', 'TADA.LatitudeMeasure', 'TADA.LongitudeMeasure',
+#' 'TADA.ResultMeasureValue', 'TADA.ComparableDataIdentifier', 'TADA.ActivityType.Flag', 
+#' and 'TADA.ActivityDepthHeightMeasure.MeasureValue'.
+#' @param type Character argument identifying which Activity Types to look for while pairing replicates
+#' to their parent samples. The default type is "QC_replicate", which includes Activity Type Codes: 
+#' "Quality Control Field Replicate Habitat Assessment",
+#' "Quality Control Field Replicate Msr/Obs",
+#' "Quality Control Field Replicate Portable Data Logger",
+#' "Quality Control Field Replicate Sample-Composite", and
+#' "Quality Control Sample-Field Replicate".
+#' @param time_window Numeric argument defining the time window in seconds within 
+#' which to search for parent samples. The default time window is 300 seconds or 5 minutes.
+#' 
+#' @return This function adds two columns to the original data frame: 'TADA.DateTime' and 
+#' 'TADA.ReplicateSampleID'. 'TADA.DateTime' combines the columns 'ActivityStartDate' and 
```

Comment:
Creating TADA.DateTime may not be necessary. See: ActivityStartDateTime. Does that give you the date time in the format you need?

Reply:
Oh, good call! I didn't notice that column before. Let me try it out and see what happens.